src/functions/utils.php

```diff
@@ -158,6 +158,30 @@ function tribe_get_request_var( $var, $default = null ) {
 	}
 }
 
+if ( ! function_exists( 'tribe_get_global_query_object' ) ) {
+	/**
+	 * Grabs the $wp_query global in a safe way with some fallbacks that help prevent fatal errors
+	 * on sites where themes or other plugins directly manipulate the $wp_query global.
+	 *
+	 * @since TBD
+	 *
+	 * @return object|boolean The $wp_query, the $wp_the_query if $wp_query empty, false otherwise.
+	 */
+	function tribe_get_global_query_object() {
+		global $wp_query;
+
+		if ( ! empty( $wp_query ) ) {
+			return $wp_query;
+		}
+
+		if ( ! empty( $_GLOBALS['wp_the_query'] ) ) {
+			return $_GLOBALS['wp_the_query'];
+		}
+
+		return false;
```

Comment:
I think in this case `null` might be a better default as `wp_query` is an object I think if the object is not present `null` should be the default representation of an object.

Reply:
Sure thing, will do.